ui/src/main/scala/org/hyperscala/ui/module/SocialSharing.scala

```diff
@@ -0,0 +1,175 @@
+      s"mailto:?${toQueryString(ld)}"
+    }
+  }
+
+  class FacebookButton(messageData: MessageData, customIcon: String = null, customClazz: Seq[String] = null)
+    extends SocialMediaButton(messageData, customIcon, customClazz) {
+    val name = "facebook"
+
+    def buildLink(md: MessageData): String = {
+      val ld = Map("u" -> md.getOrElse("shareLink", ""))
+      s"https://www.facebook.com/sharer/sharer.php?${toQueryString(ld)}"
+    }
+  }
+
+  class LinkedinButton(messageData: MessageData, customIcon: String = null, customClazz: Seq[String] = null)
+    extends SocialMediaButton(messageData, customIcon, customClazz) {
+    val name ="linkedin"
+
+    def buildLink(md: MessageData): String = {
+      val ld: LinkData = Map("mini" -> "true", "url" -> md.getOrElse("shareLink", ""),
+        "title" -> md.getOrElse("title", ""),
+      "summary" -> md.getOrElse("description", ""))
+      s"http://www.linkedin.com/shareArticle?${toQueryString(ld)}"
+    }
+  }
+
+  class TwitterButton(messageData: MessageData, customIcon: String = null, customClazz: Seq[String] = null)
+    extends SocialMediaButton(messageData, customIcon, customClazz) {
+    val name = "twitter"
+
+    def buildLink(md: MessageData): String = {
+      val ld: LinkData = Map("status" -> s"${md.getOrElse("description", "")} ${md.getOrElse("shareLink","")}")
+      s"http://twitter.com/home?${toQueryString(ld)}"
+    }
+  }
+
+  class GoogleplusButton(messageData: MessageData, customIcon: String = null, customClazz: Seq[String] = null)
+    extends SocialMediaButton(messageData, customIcon, customClazz) {
+    val name = "google_plus"
+    override def itemTagClassDesc(n: String) = super.itemTagClassDesc("googleplus")
+    override val anchorSpanContent = "google+"
+
+    def buildLink(md: MessageData): String = {
+      val ld: LinkData = Map("url" -> s"${md.getOrElse("description", "")} ${md.getOrElse("shareLink", "")}")
+      s"https://plus.google.com/share?${toQueryString(ld)}"
+    }
+  }
+
+  class PinterestButton(messageData: MessageData, customIcon: String = null, customClazz: Seq[String] = null)
+    extends SocialMediaButton(messageData, customIcon, customClazz) {
+    val name = "pinterest"
+
+    def buildLink(md: MessageData): String = {
+      //example for if "media" key was optional...
+      val base: LinkData = Map("url" -> md.getOrElse("shareLink", ""),
+        "description" -> md.getOrElse("description", ""))
+
+      val ld = md.get("imageLink") match {
+        case Some(v) => base + ("media" -> v)
+        case _ => base
+      }
+      s"http://pinterest.com/pin/create/button/?${toQueryString(ld)}"
+    }
+  }
+
+  class GithubButton(messageData: MessageData, customIcon: String = null, customClazz: Seq[String] = null)
+    extends SocialMediaButton(messageData, customIcon, customClazz) {
+    val name = "github"
+
+    def buildLink(md: MessageData): String = "https://github.com/darkfrog26/hyperscala"
```

Comment:
I understand your concern.  One would not be expected to use the `GithubButton` as is, but rather to override the `buildLink` method, which in the case of the `GithubButton`, is simply the github project one wishes to link to. 

I could replace the `buildLink` method with a link of "foo" or make it blank or something, but separating the link and the `buildLink` method for the `GithubButton` is a distinction without a difference. 

Another option might be to remove the `GithubButton` or all the buttons (`EmailButton`, `FacebookButton`, `LinkedinButton`, `TwitterButton`, `GoogleplusButton`, `PinterestButton`, and `GithubButton`) completely and simply render them in the example.   Most of the buttons have more functionality so the default implementation is reasonable for them. 

How about I remove the `GithubButton` from the `SocialSharing.scala` and implement it in the example?  That way the default is not silly and one could see from the example how to implement it.